### PR TITLE
PP-450/optimise-gas-consumption-v2

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- ~~Removing feeReceiver~~
- Changes on methods to optimise gas consumption 

## Why

- The primary objective of RIF Relay is to relay transaction the cheapest way possible.

## Refs

- [Jira](https://rsklabs.atlassian.net/browse/PP-450)

### CODE REVIEW

**Review by AndresQuijano — APPROVED** [review]

File: test/relayHub/RelayHub.test.ts [hunk 1]
```diff
@@ -945,6 +946,61 @@ contract(
                     'Relay request was properly processed'
                 );
             });
+
+            it('Should success a relayRequest without paying fee', async () => {
```
Comment (ironFe93) [inline test/relayHub/RelayHub.test.ts:950]:
```suggestion
            it('Should relay a relayRequest without paying fee', async () => {
```

File: test/relayHub/RelayHub.test.ts [hunk 2]
```diff
@@ -1172,6 +1228,85 @@ contract(
                 );
             });
 
+            it('Should success a deployRequest without paying fee', async () => {
```
Comment (ironFe93) [inline test/relayHub/RelayHub.test.ts:1231]:
```suggestion
            it('Should relay a deployRequest without paying fee', async () => {
```

File: test/relayHub/RelayHub.test.ts [hunk 3]
```diff
@@ -945,6 +946,61 @@ contract(
...
+
+                const result = relayHubInstance.relayCall(
+                    relayRequest,
+                    signature,
+                    { from: relayWorker }
+                );
+                await assert.isFulfilled(result);
+            });
+
+            it('Should success a relayRequest paying fee', async () => {
```
Comment (ironFe93) [inline test/relayHub/RelayHub.test.ts:969]:
```suggestion
            it('Should relay a relayRequest paying fee', async () => {
```

File: test/relayHub/RelayHub.test.ts [hunk 4]
```diff
@@ -1172,6 +1228,85 @@ contract(
...
+
+                const result = relayHubInstance.deployCall(
+                    deployRequest,
+                    signature,
+                    { from: relayWorker }
+                );
+                await assert.isFulfilled(result);
+            });
+
+            it('Should success a deployRequest paying fee', async () => {
```
Comment (ironFe93) [inline test/relayHub/RelayHub.test.ts:1262]:
```suggestion
            it('Should relay a deployRequest paying fee', async () => {
```

**Review by ironFe93 — APPROVED** [review]
Comments are related to test descriptions

File: test/relayHub/RelayHub.test.ts (hunk 1 above)
Comment (franciscotobar) [reply test/relayHub/RelayHub.test.ts:950]: updated

File: test/relayHub/RelayHub.test.ts (hunk 3 above)
Comment (franciscotobar) [reply test/relayHub/RelayHub.test.ts:969]: updated

File: test/relayHub/RelayHub.test.ts (hunk 4 above)
Comment (franciscotobar) [reply test/relayHub/RelayHub.test.ts:1262]: updated

**Review by antomor — APPROVED** [review]
@franciscotobar LGTM, but there's one test that is failing.

File: contracts/Collector.sol [hunk 5]
```diff
@@ -95,7 +92,7 @@ contract Collector is ICollector{
     onlyOwner
     {
         uint256 balance = token.balanceOf(address(this));
-        require(balance > partners.length, "Not enough balance to split");
+        require(balance >= partners.length, "Not enough balance to split");
```
Comment (antomor) [inline contracts/Collector.sol:95]: The new change isn't covered by any test

**Pull request merged**